examples/object/testhooks/validate-ssa.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+set -aeuo pipefail
+
+# This script is used to validate the ssa feature, triggered by the
+# uptest framework via `uptest.upbound.io/post-assert-hook`: https://github.com/crossplane/uptest/tree/e64457e2cce153ada54da686c8bf96143f3f6329?tab=readme-ov-file#hooks
+
+LABELER_OBJECT="examples/object/object-ssa-labeler.yaml"
+${KUBECTL} apply -f ${LABELER_OBJECT}
+${KUBECTL} wait -f ${LABELER_OBJECT} --for condition=ready --timeout=1m
+
+if ! ${KUBECTL} get service sample-service -o jsonpath='{.metadata.annotations}' | grep -v "last-applied-configuration"; then # This annotation should not be present when SSA is enabled
+  echo "SSA validation failed! Annotation 'last-applied-configuration' should not exist when SSA is enabled!"
+  #exit 1
+fi
+if ! (${KUBECTL} get service sample-service -o jsonpath='{.metadata.labels.some-key}' | grep -q "some-value" && ${KUBECTL} get service sample-service -o jsonpath='{.metadata.labels.another-key}' | grep -q "another-value"); then
+  echo "SSA validation failed! Labels 'some-key' and 'another-key' from both Objects should exist with values 'some-value' and 'another-value' respectively!"
+  #exit 1
+fi
+echo "Successfully validated the SSA feature!"
```

Comment:
is this the only e2e test we have? 😱

Reply:
Currently, we rely on the uptest framework for e2e tests just like other providers and what we can do with uptest is limited, but still better than nothing. Also due to some current [limitations](https://github.com/crossplane/uptest/issues/17) that is about to be solved, we couldn't really integrate this test into the CI.

I would suggest:
- Merging this PR by manually triggering those tests.
- Fast follow to enable them in CI with the corresponding uptest fix, so that we can make sure they run for every PR.
- Creating a ticket to discuss whether we need a more sophisticated testing framework here.